ledger/src/blockstore_processor.rs

```diff
@@ -190,13 +190,16 @@ pub fn execute_batch<'a>(
     let pre_commit_callback = |_timings: &mut _, processing_results: &_| -> PreCommitResult {
         match extra_pre_commit_callback {
             None => {
+                // We're entering into one of the block-verification methods.
                 get_first_error(batch, processing_results)?;
                 Ok(None)
             }
             Some(extra_pre_commit_callback) => {
-                // We're entering into the block-producing unified scheduler special case...
+                // We're entering into the block-production unified scheduler special case...
                 // `processing_results` should always contain exactly only 1 result in that case.
-                assert_eq!(processing_results.len(), 1);
+                let [result] = processing_results else {
+                    panic!("unexpected result count: {}", processing_results.len());
+                };
                 assert!(transaction_indexes.is_empty());
```

Comment:
could we add a descriptive string on why this should be empty? IIRC BP variant should not have indexes, so something along the lines of "block-production path should not have batch transaction indexes"

Reply:
> IIRC BP variant should not have indexes, so something along the lines of "block-production path should not have batch transaction indexes"

Actually, this is true only if we're ignoring `solana-test-validator`....

how about this? I added x-refs: bd953be15e39ac7c77d94f0b8f076815f6895022

hopefully, this will guide the right place of existing longer prose for correct information...